openchain/consensus/obcpbft/obc-sieve.go

```diff
@@ -317,6 +339,51 @@ func (op *obcSieve) processExecute() {
 		op.id, verify.BlockNumber)
 	op.broadcastMsg(&SieveMessage{&SieveMessage_Verify{verify}})
 	op.recvVerify(verify)
+
+	op.pbft.unlock()
+
+	logger.Debug("Sieve replica %d waiting for result decision on block %d", op.id, seqNo)
+	select {
+	case result := <-op.validResultChan:
```

Comment:
Before this patch, the sieve flow worked roughly as:
 1. Begin Sieve Request (Sieve Message thread)
 2. BeginBatch (Sieve Message thread)
 3. ExecTxs (Sieve Message thread)
 4. PreviewBatch (Sieve Message thread)
 5. Send Verify (Sieve Message thread)
 6. Assign final ordering (PBFT Message thread)
 7. Queue execution (PBFT Message thread)
 8. Make decision (Execution thread)
 9. CommitBatch (Execution thread)

Although the flow is certainly simpler, the problem here is that the Batch, Exec, and Preview operations will all happening on the Sieve Message Thread, while the Commit is happening via the Execution thread.  The new flow looks as follows:

 1. Begin Sieve Request (Sieve Message thread)
 2. BeginBatch (Execution thread)
 3. ExecTxs (Execution thread)
 4. PreviewBatch (Execution thread)
 5. Send Verify (Execution thread)
 6. Assign final ordering (PBFT Message thread)
 7. Queue execution (PBFT Message thread)
 8. Make decision (PBFT Message thread)
 9. CommitBatch (Execution thread)

This is another effect of attempting to view the `Executor` as a network service, that executions are queued asynchronously, and the `Orderer` can only interact with their results.